homeassistant/components/meteo_france/sensor.py

```diff
@@ -79,9 +79,10 @@ def __init__(self, sensor_type: str, coordinator: DataUpdateCoordinator):
         """Initialize the Meteo-France sensor."""
         super().__init__(coordinator)
         self._type = sensor_type
-        city_name = self.coordinator.data.position["name"]
-        self._name = f"{city_name} {SENSOR_TYPES[self._type][ENTITY_NAME]}"
-        self._unique_id = f"{self.coordinator.data.position['lat']},{self.coordinator.data.position['lon']}_{self._type}"
+        if hasattr(self.coordinator.data, "position"):
```

Comment:
This fix doens't match your comment here? https://github.com/home-assistant/core/issues/39935#issuecomment-691050174

Reply:
Yes it does.

Before #39432 (`CoordinatorEntry`), `super().__init__(...)` was missing on `class MeteoFranceAlertSensor(MeteoFranceSensor)`

After #39432 (`CoordinatorEntry`): the `MeteoFranceAlertSensor` super calls `MeteoFranceSensor` init. So it tries to get the name of the city via `position`, but it cannot because the alert is about a department and has no `position` attribute.

An other solution would be to no heritate from `MeteoFranceSensor` and retreive common attributes.

Reply:
Other solution again:
Store the city and departement infos in the data or config, so we don't get them from the fetch